### PR TITLE
[e2e] Send ETH to an EOA using Ganache network on Android

### PR DESCRIPTION
**Pull request by seaona**
**Description**
This PR adds an e2e testcase for Android with wdio/appium, using the Ganache network. The test is a Simple send flow to an EOA. For that, it's done the following:

- A Confirmations folder has been created for including the android e2e tests related to Confirmations
- A Confirmations suite has been created in order to run all confirmation related tests by using the flag `--suite confirmations`
- A Ganache server will be started whenever we use `Ganache server is started` from common steps
- The Ganache server will be stopped whenever we use `Ganache server is stopped` from common steps
- Ganache sets `10000` ETH by default to the generated accounts given the seedphrase we pass it. This seedphrase is defined on ` Accounts.getValidAccount()`
  - It seems that the envar `MM_TEST_ACCOUNT_SRP` is not working and not correctly set. For this reason, running local tests we can add a seedphrase, in case the envar does not work  ` seedPhrase: process.env.MM_TEST_ACCOUNT_SRP || 'phrase upgrade clock rough situate wedding elder clever doctor stamp excess tent',`. Seems this is related to issue 762

**Manual testing steps**
- Connect an Android device to your computer / use simulator (see Android setup on readme)
  - You'll need to set the following vars
```
export ANDROID_HOME=/home/PATH_TO_ANDROID/Android/Sdk
export ANDROID_SDK_ROOT=/home/PATH_TO_ANDROID/Android/Sdk
```

- Open a terminal
- Expose the needed ports on your Android device
  - `adb reverse tcp:8081 tcp:8081`
  - `adb reverse tcp:8545 tcp:8545`  -- where ganache server runs
- Create the QA build
  - `yarn setup`
  - `yarn start:android:qa` 
- Open another terminal
  - run yarn watch 
- On the first terminal run the specific testcase `yarn test:wdio:android --spec wdio/features/Confirmations/SendEthEOA.feature`
- Notice you might need to add a seedphrase on the envar file and if it does not work, you can add it directly as indicated above
  
**Screenshots/Recordings**

Screenshot from 2023-04-19 10-56-53

[android-ganache-send-eth.webm](https://user-images.githubusercontent.com/54408225/233032738-0592c951-b283-433c-92b3-8c2d125f38ef.webm)

**Issue**

- Resolves https://github.com/MetaMask/mobile-planning/issues/896

**Checklist**

* [X] There is a related GitHub issue
* [X] Tests are included if applicable
* [X] Any added code is fully documented

### CODE REVIEW

File: wdio.conf.js [hunk 1]
```diff
@@ -28,6 +32,10 @@ export const config = {
   //
   specs: ['./wdio/features/*.feature', './wdio/features/**/*.feature'],
 
+  suites: {
+    confirmations: ['./wdio/features/Confirmations/*.feature']
+  },
+
```
Comment (seaona) [inline wdio.conf.js:38]: Added a `confirmations` test suite, so we can easily test all confirmations tests passing the flag `--suite confirmations`

File: wdio.conf.js [hunk 2]
```diff
@@ -277,8 +285,11 @@ export const config = {
    * @param {String}                   uri      path to feature file
    * @param {GherkinDocument.IFeature} feature  Cucumber feature object
    */
-  // beforeFeature: function (uri, feature) {
-  // },
+   beforeFeature: async function (uri, feature) {
+    if(uri.includes('Confirmations')) {
+      await ganacheServer.start({ mnemonic: validAccount.seedPhrase });
+    }
+   },
```
Comment (seaona) [inline wdio.conf.js:292]: ganache server will start on every feature file inside `Confirmations`

File: wdio/features/Confirmations/SendEth.feature [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Feature: Sending ETH to an Externally Owned Account (EOA)
+  A user should be able to send ETH to another EOA address.
+
+  Scenario: Import wallet
+    Given the app displayed the splash animation
+    And I have imported my wallet
+    And I tap No Thanks on the Enable security check screen
+    And I tap No thanks on the onboarding welcome tutorial
+
+  Scenario: Setting up Ganache local network
```
Comment (seaona) [inline wdio/features/Confirmations/SendEth.feature:14]: This scenario should be injected in the state in the future

File: wdio/features/Confirmations/SendEth.feature [hunk 4]
```diff
@@ -0,0 +1,56 @@
+@androidApp
+@ChainScenarios
+@regression
+
+Feature: Sending ETH to an Externally Owned Account (EOA)
+  A user should be able to send ETH to another EOA address.
+
+  Scenario: Import wallet
```
Comment (seaona) [inline wdio/features/Confirmations/SendEth.feature:8]: This scenario should be injected in the state in the future

File: wdio/features/Confirmations/SendEthEOA.feature [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+Feature: Sending ETH to an EOA
+  A user should be able to send ETH to another EOA address.
+
+  Scenario: Import wallet
+    Given the app displayed the splash animation
+    And I have imported my wallet
+    And I tap No Thanks on the Enable security check screen
+    And I tap No thanks on the onboarding welcome tutorial
+
+  Scenario: Setting up Ganache local network
```
Comment (seaona) [inline wdio/features/Confirmations/SendEthEOA.feature:14]: This scenario should be injected in the state in the future

File: wdio/features/Confirmations/SendEthEOA.feature [hunk 6]
```diff
@@ -0,0 +1,57 @@
+@androidApp
+@ChainScenarios
+@regression
+
+Feature: Sending ETH to an EOA
+  A user should be able to send ETH to another EOA address.
+
+  Scenario: Import wallet
```
Comment (seaona) [inline wdio/features/Confirmations/SendEthEOA.feature:8]: This scenario should be injected in the state in the future

**Review by jpuri — APPROVED** [review]

File: wdio/features/Confirmations/SendEthEOA.feature [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    Given the app displayed the splash animation
+    And I have imported my wallet
+    And I tap No Thanks on the Enable security check screen
+    And I tap No thanks on the onboarding welcome tutorial
```
Comment (gantunesr) [inline wdio/features/Confirmations/SendEthEOA.feature:12]: Nit: should it be `No Thanks` or `No thanks`?

File: wdio/features/Confirmations/SendEthEOA.feature [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+  Scenario: Setting up Ganache local network
+    Given Ganache server is started
+    Given I tap on the burger menu
+    And I tap on "Settings" in the menu
+    And In settings I tap on "Networks"
+    And I tap on the Add Network button
```
Comment (gantunesr) [inline wdio/features/Confirmations/SendEthEOA.feature:19]: Just curiosity for the rule here, why you put `"Settings"` in quotation marks but not `Add Network`?

File: wdio/features/Confirmations/SendEthEOA.feature [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    When On the Main Wallet view I tap "ETHER"
+    And On the Main Wallet view I tap "Send"
+    And I enter address "<Address>" in the sender's input box
+    When I tap button "Next" on Send To view
+    Then I proceed to the amount view
+
+    When I type amount "<Amount>" into amount input field
+    And I tap button "Next" on the Amount view
+    Then I should be taken to the transaction confirmation view
+    And the token amount <Amount> to be sent is visible
```
Comment (gantunesr) [inline wdio/features/Confirmations/SendEthEOA.feature:48]: Nit: Is it right to have `"<Amount>"` and `<Amount>`?

**Review by gantunesr — COMMENTED** [review]
LGTM, some minor comments

File: wdio/features/Confirmations/SendEthEOA.feature (hunk 9 above)
Comment (Andepande) [reply wdio/features/Confirmations/SendEthEOA.feature:48]: There's some inconsistency in how some steps are paremeterised, going forward we should just standardise it

File: wdio/features/Confirmations/SendEthEOA.feature (hunk 8 above)
Comment (Andepande) [reply wdio/features/Confirmations/SendEthEOA.feature:19]: There's some inconsistency in how some steps are parameterised, going forward we should just standardise it

**Review by Andepande — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
This looks good to me 🌮 🌮

File: wdio/features/Confirmations/SendEthEOA.feature (hunk 7 above)
Comment (seaona) [reply wdio/features/Confirmations/SendEthEOA.feature:12]: @gantunesr it's a good observation. In this PR I've just re-utilized the steps defined in `step-definitions` folder. In this case, this step is set as lower case `thanks`

https://github.com/MetaMask/metamask-mobile/blob/main/wdio/step-definitions/common-steps.js#L111

However I do agree we should try to consolidate them, but for the purpose of this PR I just wanted to touch the bare minimum to implement Ganache. We could do this in a separate PR, as all test would then need to be updated to the new statement

File: wdio/features/Confirmations/SendEthEOA.feature (hunk 8 above)
Comment (seaona) [reply wdio/features/Confirmations/SendEthEOA.feature:19]: Thank you for your comments @gantunesr @Andepande . 
In this case, we are referring to the specific step `I tap on the Add Network button` defined here
https://github.com/MetaMask/metamask-mobile/blob/main/wdio/step-definitions/add-networks.steps.js#L101

On the `"Settings"` case, we are calling another step definition, and in this case we pass `Settings` as a variable, that's why it's done this way, see here:
https://github.com/MetaMask/metamask-mobile/blob/main/wdio/step-definitions/add-networks.steps.js#L78

As in the comment above, I very much agree this is not consolidated and we should work on a PR for consolidating all tests. However, this should be out of the scope of this PR.

File: wdio/features/Confirmations/SendEthEOA.feature (hunk 9 above)
Comment (seaona) [reply wdio/features/Confirmations/SendEthEOA.feature:48]: thank you again for bringing this up! Same comment above applies here @gantunesr @Andepande

**Review by gantunesr — APPROVED** [review]

File: wdio/features/Confirmations/SendEthEOA.feature [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+  Scenario: Import wallet
+    Given the app displayed the splash animation
+    And I have imported my wallet
+    And I tap No Thanks on the Enable security check screen
+    And I tap No thanks on the onboarding welcome tutorial
+
+  Scenario: Setting up Ganache local network
+    Given Ganache server is started
+    Given I tap on the burger menu
```
Comment (cortisiko) [inline wdio/features/Confirmations/SendEthEOA.feature:16]:
```suggestion
    When I tap on the burger menu
```

**Review by cortisiko — COMMENTED** [review]
minor comment

**Pull request merged**